### PR TITLE
Enable install-powershell.ps1 to use MSI

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

Enable `install-powershell.ps1` to install the MSI and not just the zip on Windows.  Also fixed it so it works from Windows PowerShell.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by PaulHigin — APPROVED** [review]

File: tools/install-powershell.ps1 [hunk 1]
```diff
@@ -51,8 +54,19 @@ if (-not $Destination) {
         $Destination = "${Destination}-daily"
     }
 }
-$Destination = $PSCmdlet.SessionState.Path.GetUnresolvedProviderPathFromPSPath($Destination)
-Write-Verbose "Destination: $Destination" -Verbose
+
+# Windows PowerShell doesn't have $PSCmdlet, so we'll just use the given path as-is
```
Comment (vexx32) [inline tools/install-powershell.ps1]: Are you sure about that?

```powershell
# Win 5.1
function test {
>> [cmdletbinding()]
>> param()
>> $PSCmdlet | fl *
>> }
PS Z:\> test


ParameterSetName     : __AllParameterSets
MyInvocation         : System.Management.Automation.InvocationInfo
PagingParameters     :
InvokeCommand        : System.Management.Automation.CommandInvocationIntrinsics
Host                 : System.Management.Automation.Internal.Host.InternalHost
SessionState         : System.Management.Automation.SessionState
Events               : System.Management.Automation.PSLocalEventManager
JobRepository        : System.Management.Automation.JobRepository
JobManager           : System.Management.Automation.JobManager
InvokeProvider       : System.Management.Automation.ProviderIntrinsics
Stopping             : False
CommandRuntime       : test
CurrentPSTransaction :
CommandOrigin        : Runspace
```

Comment (SteveL-MSFT) [reply tools/install-powershell.ps1]: Hmm, this initially failed in my Windows PowerShell 5.1 console.  Let me see what's going on here.

Comment (SteveL-MSFT) [reply tools/install-powershell.ps1]: Just tried it and it seems fine, so don't know what happened previously.  Thanks for calling this out!

File: tools/install-powershell.ps1 [hunk 2]
```diff
@@ -141,27 +156,43 @@ try {
         Write-Verbose "About to download package from '$downloadURL'" -Verbose
 
         $packagePath = Join-Path -Path $tempDir -ChildPath $packageName
+        if ($PSVersionTable.PSEdition -eq "Desktop") {
+            # On Windows PowerShell, progress can make the download significantly slower
+            $oldProgressPreference = $ProgressPreference
```
Comment (adityapatwardhan) [inline tools/install-powershell.ps1]: `Invoke-WebRequest` can throw terminating error, so I would prefer resetting the `ProgressPreference` in a try-finally block.

File: tools/install-powershell.ps1 [hunk 3]
```diff
@@ -141,27 +156,43 @@ try {
...
+            $ProgressPreference = $oldProgressPreference
+        }
+
         $contentPath = Join-Path -Path $tempDir -ChildPath "new"
 
         New-Item -ItemType Directory -Path $contentPath > $null
         if ($IsWinEnv) {
-            Expand-Archive -Path $packagePath -DestinationPath $contentPath
+            if ($UseMSI) {
+                Start-Process $packagePath -Wait
```
Comment (adityapatwardhan) [inline tools/install-powershell.ps1]: Should we also check for elevation?

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Review by adityapatwardhan — APPROVED** [review]

File: tools/install-powershell.ps1 [hunk 4]
```diff
@@ -26,6 +26,9 @@ param(
     [Parameter()]
     [switch] $Daily,
 
+    [Parameter()]
+    [switch] $UseMSI,
```
Comment (TravisEz13) [inline tools/install-powershell.ps1]: `UseMSI` is ignored with the `Daily` Switch, the two should be on different parameter sets

Comment (TravisEz13) [reply tools/install-powershell.ps1]: Similar issue with `destination`, `AddToPath`, `DoNotOverwrite`, but I find this less confusing in how it is currently implemented.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: tools/install-powershell.ps1 (hunk 4 above)
Comment (SteveL-MSFT) [reply tools/install-powershell.ps1]: Fixed

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
Consider adding a test to https://github.com/PowerShell/PowerShell/blob/master/.vsts-ci/install-ps.yml in the future.

**Pull request merged**